src/main/java/org/swisspush/redisques/util/QueueStatisticsCollector.java

```diff
@@ -45,28 +49,79 @@
     private final Map<String, AtomicLong> queueFailureCount = new HashMap<>();
     private final Map<String, Long> queueBackpressureTime = new HashMap<>();
     private final Map<String, Long> queueSlowDownTime = new HashMap<>();
-
+    private final ConcurrentMap<String, AtomicLong> queueMessagePaceCtr = new ConcurrentHashMap<>();
+    private volatile Map<String, Long> queueMessagePace = new HashMap<>();
     private final RedisAPI redisAPI;
     private final LuaScriptManager luaScriptManager;
     private final String queuePrefix;
+    private final Vertx vertx;
 
     public QueueStatisticsCollector(RedisAPI redisAPI, LuaScriptManager luaScriptManager,
-            String queuePrefix) {
+            String queuePrefix, Vertx vertx, int paceIntervalTime) {
         this.redisAPI = redisAPI;
         this.luaScriptManager = luaScriptManager;
         this.queuePrefix = queuePrefix;
+        this.vertx = vertx;
+        paceStatisticsScheduler(paceIntervalTime);
     }
 
     /**
-     * Does reset all statistics values of the given queue. In memory but as well the persisted
+     * This scheduled task collects and calculates the message pace per queue in the local system
```

Comment:
What is exactly the "message pace", there is also the term "queues pace". I think it is the same. Should we stick to one term and further describe / define it?

Reply:
OK, I think it makes sense, hope that it doesn't take too much time to adapt all the code.